broker/broker.go

```diff
@@ -414,56 +414,64 @@ func (b *CdnServiceBroker) Update(
 	return brokerapi.UpdateServiceSpec{IsAsync: provisioningAsync}, nil
 }
 
-// createBrokerOptions will attempt to take raw json and convert it into the "Options" struct.
-func (b *CdnServiceBroker) createBrokerOptions(details []byte) (options Options, err error) {
-	if len(details) == 0 {
-		err = errors.New("must be invoked with configuration parameters")
-		return
-	}
+// parseProvisionDetails will attempt to parse the update details and then verify that BOTH least "domain" and "origin"
+// are provided.
+func (b *CdnServiceBroker) parseProvisionDetails(details brokerapi.ProvisionDetails) (options Options, err error) {
```

Comment:
I think typically we avoid specifying the names of the return variables in the signature, instead opting for explicit `return foo, bar` everywhere. That said, I'm not sure we have a set convention for it.